tests/test_integration.py

```diff
@@ -343,17 +349,19 @@ def test_interpreter_constraints_to_pex_info_py2():
     assert set(['>=2.7', '<3']) == set(pex_info.interpreter_constraints)
 
 
-@pytest.mark.skip('https://github.com/pantsbuild/pex/issues/470')
 def test_interpreter_constraints_to_pex_info_py3():
-  with temporary_dir() as output_dir:
-    # target python 3
-    pex_out_path = os.path.join(output_dir, 'pex_py3.pex')
-    res = run_pex_command(['--disable-cache',
-      '--interpreter-constraint=>3',
-      '-o', pex_out_path])
-    res.assert_success()
-    pex_info = get_pex_info(pex_out_path)
-    assert ['>3'] == pex_info.interpreter_constraints
+  py3_interpreter = ensure_python_interpreter('3.6.3')
```

Comment:
rather than mocking out `os.getenv` here, how about just temporarily setting the `PATH` env var with `twitter.common.contextutil.environment_as` for the scope of this test?